### ISSUE TITLE
Keycodes being sent as a single uint8

### ISSUE THREAD

**Issue by ToastyStoemp** [the-via/app#102]
After playing around with the custom menu's we've been trying to push the capabilities of what you an do to the limits.
The custom KeyCode field allows for a keycode or hex value to be sent. Though we quickly noticed that only the last 2 'characters' were being sent. 
For example hex 0xABCD will result in CD arriving on the keyboard.

So we dove a little deeper. 
As far as we understand, VIA keycodes communication is only sending 1 unsigned int 8 byte over HID. While all keycodes in within QMK are defined as unsigned int 16. Though only the 'special' keycodes go above the 255 threshold.
I understand that this is quite a fundamental change to the inner workings of VIA, but wouldn't it be an improvement of the project to always send 2 uint8 bytes, even if one is redundant?

Alternatively limiting the custom keycode ui element to only 2 characters (for example 0xAB). But making a hex and or string field for custom menu's available instead. This alternative custom menu item will not be checked against keycodes, but only be used for data transfer. Which could be useful for setting exact colors, or text on an OLED screen.

**Comment by wilba** [comment]
This is a bug, it should be sending a 16-bit keycode. Will be fixed ASAP.

**Comment by Cipulot** [comment]
After some playing it seems like the range control has the same behavior.

As a test-code I implemented a slider with values ranging from `0` to `1023`. In the firmware I gathered the data and displayed it in the HID console. Since the data is split into `uint8_t` chunks I applied bitwise operations to obtain a single `uint16_t`. To be noted also that the order of the HI and LO uint8_t seems to differ compared to the keycode shifting.

In the keycode example we used `value_data[1] | (value_data[0] << 8)`, but doing the same in the range operation results in a shift of 256. Therefore to obtain the true 0 (might be dependent on the wrongly structured data being sent tho) we have to use `value_data[0] | (value_data[1] << 8)`.

By using the later I get rid of the offset but the data being set is still only of `uint8_t` size, like it's visible from this log:
<img width="886" alt="Screenshot 2023-01-20 at 4 35 24 PM" src="https://user-images.githubusercontent.com/40441626/213742355-6d9d98d5-c1e1-45ea-b4bc-f260e8f5e614.png">

Note that in the above data0 is `value_data[0]`, data1 is `value_data[1]`and the final number is the result of the bitwise operation printed from a uint16_t.

Tested by both me and @ToastyStoemp. At the same time we tested the patches you pushed and waiting for merging for the keycodes and proved to be good and working. Ideally a similar patch here is needed. It would be also preferable to use the same shifting, so that the same bitwise operation can be performed.
